go/pkg/client/bytestream.go

```diff
@@ -83,7 +83,7 @@ func (c *Client) ReadResourceToFile(ctx context.Context, name, fpath string) (in
 }
 
 func (c *Client) readToFile(ctx context.Context, name string, fpath string) (int64, error) {
-	f, err := os.Create(fpath)
+	f, err := os.OpenFile(fpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, c.RegularMode)
```

Comment:
Note that this is still changing permission from 0666 to 0644
https://cs.opensource.google/go/go/+/master:src/os/file.go;l=319;drc=6b420169d798c7ebe733487b56ea5c3fa4aab5ce